app/builders/messages/instagram/message_builder.rb

```diff
@@ -112,12 +116,9 @@ def message_params
 
   def already_sent_from_chatwoot?
     cw_message = conversation.messages.where(
-      source_id: nil,
-      message_type: 'outgoing',
-      content: message_content,
-      private: false,
-      status: :sent
+      source_id: @messaging[:message][:mid]
     ).first
+
     cw_message.update(content_attributes: content_attributes) if cw_message.present?
```

Comment:
Is updating the content_attributes of the message intentional? Could we add a comment here on why we need to do this?

Reply:
It's not necessary now since we are storing the same in the message source id.
Removing the same.